src/protocol.cpp

```diff
@@ -199,3 +199,27 @@ const std::vector<std::string> &getAllNetMessageTypes()
 {
     return allNetMessageTypesVec;
 }
+
+std::string serviceFlagToStr(const uint64_t mask, const int bit)
```

Comment:
Seems it would be enough to only pass in `bit`, or what am I missing?

Reply:
This avoid recalculating `mask`

Reply:
That's one bit shift `1ull << bit;`. Given how rarely this function is called that seems like an over-optimization.